### ISSUE TITLE
OpenAI compatible API support

### ISSUE THREAD

**Issue by Zane-XY** [jeremychone/rust-genai#5]
I noticed that the url in this crate is hardcoded, which currently only supports the official API endpoints of each AI service provider.

Would there be a plan to make the endpoint configurable? For example, allowing users to specify Azure OpenAI endpoints through configuration would greatly enhance flexibility.

**Comment by jeremychone** [comment]
@Zane-XY yes, endpoints will be configurable per adapter kind. I need to find the right way to do it (e.g., host/port v.s path ...)

**Comment by jeremychone** [comment]
@Zane-XY btw, feel free to explain your particular usecase. I will make sure it get covered.

**Comment by Zane-XY** [comment]
In my use case, the service url and models are different, but the service is OpenAI compatible. Really appreciated the fast response!

**Comment by jeremychone** [comment]
@Zane-XY thanks. Is this aws bedrock / Google vertexai, or a custom service somewhere ?
Also, is it a Ollama server? (their OpenAi compatibility layer requires some custom behaviors)

**Comment by Zane-XY** [comment]
It's an enterprise hosted AI service.

**Comment by jeremychone** [comment]
Ok, that's will probably be a Custom Adapter then. I will get to it, genai should support this usecase.

**Comment by Boscop** [comment]
I'm using Jan.ai, TabbyML and LM Studio to run local models with local API server exposing an OpenAI-compatible API.
I would like to use this crate to make requests to them (also for embeddings) 🙂

**Comment by InAnYan** [comment]
Hi! + for this feature.

Basically, it would be better to just make API base URL to be a variable that can be changed in constructor.

That's what I did to use ollama (it was not `genai`, other project).

+ You can also make a mock server and test the code. IDK why and where, but you can make a mock webserver and it can give you a URL which you can supply to custom code

**Comment by yihong0618** [comment]
seems it supported now;
e.g.
https://github.com/jeremychone/rust-genai/blob/main/examples/c06-target-resolver.rs

**Comment by jeremychone** [comment]
@InAnYan yes, sorry for the delay all. This was fixed with the `ServiceTargetResolver`

**Comment by InAnYan** [comment]
Thanks! Great to see that `genai` is actively developed
